Gems/Atom/RPI/Code/Source/RPI.Private/RPISystemComponent.cpp

```diff
@@ -75,6 +75,8 @@ namespace AZ
             m_materialFunctorRegistration = aznew MaterialFunctorSourceDataRegistration;
             m_materialFunctorRegistration->Init();
         #endif
+
+            XRRegisterInterface::Register(this);
```

Comment:
Trick: You can avoid calling the interface Register/Unregister in the constructor/destructor if instead of inheriting from `public IXRRegisterInterface` the class inherits from `public XRRegisterInterface::Registrar`. It does it for you :)